### PR TITLE
Add docstring in vm_manage module

### PR DESCRIPTION
**Pull request by khomesh24**
Signed-off-by: khomesh24 <khomeshthakre24@gmail.com>
Add docstring in vm_manage module

### CODE REVIEW

File: satellite/modules/vm_manage.py [hunk 1]
```diff
@@ -10,6 +10,18 @@
...
+
+    :param compute_ip: Remote system IP address
+    :param name: Name of virtual machine
+    :param ram: RAM size for virtual machine
+    :param cpus: Number of virtual CPUS for virtual machine
+    :param disk_size: Disk size for virtual machine
+    :param location_url: URL location of OS
+    :param kickstart_loc: location of kickstart
+
+    :returns: Boolean, true if success or false
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: change returns line to Boolean **True** if success else **False**

File: satellite/modules/vm_manage.py [hunk 2]
```diff
@@ -31,20 +49,39 @@ def isOnline(host):
 
 
 def virsh_start_vm(vm_name, com_ip):
+    """Starts the virtual machine on remote system
+    :param vm_name: Name of virtual machine
+    :param com_ip: Compute IP on which virtual machine is running
+
+    :return: start_vm_flag
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: above you have written **returns** here you wrote **return**
choose one

File: satellite/modules/vm_manage.py [hunk 3]
```diff
@@ -85,6 +138,20 @@ def get_memory(compute_ip, vm_name, vm_ip):
 
 
 def vm_details(compute_name, compute_ip, vm_id):
+    """Find details of virtual machine
+
+    In this function virtual machine details like ID, name, state(Running or shut),
+    virtual CPUS, Total memory allocated, Free memory, vm's IP address, vm's MAC address,
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: dont write just **vm**
it should be **CPUs**

File: satellite/modules/vm_manage.py [hunk 4]
```diff
@@ -85,6 +138,20 @@ def get_memory(compute_ip, vm_name, vm_ip):
...
+    Cumpute name( on which it is provisioned).
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: what is **Cumpute**?

File: satellite/modules/vm_manage.py [hunk 5]
```diff
@@ -85,6 +138,20 @@ def get_memory(compute_ip, vm_name, vm_ip):
...
+
+    :param compute_name: Name of compute on which vm is running
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: change **vm** to **virtual machine**

File: satellite/modules/vm_manage.py [hunk 6]
```diff
@@ -85,6 +138,20 @@ def get_memory(compute_ip, vm_name, vm_ip):
...
+    :param compute_ip: IP address of compute on which vm is running
+    :param vm_id: UUID of virtual machine
+
+    :return details: Dictionary of ID, name, state(Running or shut),
+    virtual CPUS, Total memory allocated, Free memory, vm's IP address, vm's MAC address,
+    Cumpute name
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: **Cumpute**?

File: satellite/modules/vm_manage.py [hunk 7]
```diff
@@ -109,13 +176,27 @@ def vm_details(compute_name, compute_ip, vm_id):
...
     package_info = package_info[2:]
     return package_info
 
 
 def get_repo(activation_name):
+    """Get repo list included in Activation name
+
+    :param activation_name: Name of activation
+
+    :return repo: dictionary of repo name and repo URL included in that activation
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: if you start writing the first character in uppercase follow that everywhere.
**d**ictionary is in lowercase

File: satellite/modules/vm_manage.py [hunk 8]
```diff
@@ -148,6 +237,14 @@ def get_status(compute_name, compute_ip, vm_name):
 
 
 def filter_repo(repo_info):
+    """Filter the repos
+
+    Remove all unnecessary data from repos
+
+    :param repo_info: Raw repo data
+
+    :return repo_info: cleaned repo data
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: **c**leaned is in lowercase

File: satellite/modules/vm_manage.py [hunk 9]
```diff
@@ -156,6 +253,16 @@ def filter_repo(repo_info):
 
 
 def get_vm_repo(compute_ip, vm_ip, vm_name):
+    """Get virtual machine repo
+
+    Find repo add in virtual machine and its status whether its enable or disable
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: cannot understand what **Find repo add** means

Comment (yadneshk) [inline satellite/modules/vm_manage.py]: use **repository** instead of **repo**

File: satellite/modules/vm_manage.py [hunk 10]
```diff
@@ -215,6 +330,16 @@ def vm_status(compute_ip, vm_name, vm_ip):
 
 
 def change_repo(compute_ip, vm_ip, repo_id, repo_flag, vm_name):
+    """Change the repo status
+
+    :param compute_ip: IP address of compute on which virtual machine is running
+    :param vm_ip: IP address of virtual machine
+    :param repo_id: repo ID
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: **r**epo id is in lowercase

File: satellite/modules/vm_manage.py [hunk 11]
```diff
@@ -215,6 +330,16 @@ def vm_status(compute_ip, vm_name, vm_ip):
...
+    :param repo_flag: flag of repo (enable or disable)
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: *f**lag is in lowercase

File: satellite/modules/vm_manage.py [hunk 12]
```diff
@@ -215,6 +330,16 @@ def vm_status(compute_ip, vm_name, vm_ip):
...
+    :param vm_name: Name of virtual machine
+
+    :return : success or failed
```
Comment (yadneshk) [inline satellite/modules/vm_manage.py]: **s**uccess is in lowercase

File: satellite/modules/vm_manage.py (hunk 2 above)
Comment (CURIOUSER005) [reply satellite/modules/vm_manage.py]: use return instead of returns

File: satellite/modules/vm_manage.py (hunk 1 above)
Comment (CURIOUSER005) [reply satellite/modules/vm_manage.py]: use return instead of returns

File: satellite/modules/vm_manage.py (hunk 4 above)
Comment (CURIOUSER005) [reply satellite/modules/vm_manage.py]: make it Compute Name (on which it is provisioned)

File: satellite/modules/vm_manage.py [hunk 13]
```diff
@@ -109,13 +176,27 @@ def vm_details(compute_name, compute_ip, vm_id):
...
     vm_ip = vm_ip.split("/")[0]
     package_info = os.popen("ssh root@" + compute_ip + " 'sshpass -p " + root_passwd + " ssh -o StrictHostKeyChecking=no root@" + vm_ip + " rpm -qa'").readlines()
     package_info = package_info[2:]
     return package_info
 
 
 def get_repo(activation_name):
+    """Get repo list included in Activation name
+
+    :param activation_name: Name of activation
```
Comment (CURIOUSER005) [inline satellite/modules/vm_manage.py]: I think you should add key with activation 
like activation key

**Pull request merged**